core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -1280,7 +1281,18 @@ private boolean joinDynamicFilteringExpression(Expression expression, Collection
                 }
                 comparison = (ComparisonExpression) expression;
             }
-            return comparison.getLeft() instanceof SymbolReference && comparison.getRight() instanceof SymbolReference;
+
+            return dynamicFilteringApplicable(comparison, leftSymbols, rightSymbols);
+        }
+
+        // sub expression that uses right source has to be a symbol (left can be an arbitrary expression)
+        private boolean dynamicFilteringApplicable(ComparisonExpression comparison, Collection<Symbol> leftSymbols, Collection<Symbol> rightSymbols)
```

Comment:
that name is not meaningful. Either inline this method or find better name for it (maybe, `isBuildSideExpressionSymbolReference`)

Reply:
I like the idea of inlining to avoid finding a good name...